src/rtos/FreeRTOS.c

```diff
@@ -711,7 +716,7 @@ static int freertos_update_threads(struct rtos *rtos)
 			strcpy(rtos->thread_details[tasks_found].thread_name_str, tmp_str);
 			rtos->thread_details[tasks_found].exists = true;
 
-			if (value->tcb == px_current_tcb) {
+			if (value->tcb == px_current_tcb && rtos->current_thread != 1) {
```

Comment:
Can you make a constant for FREERTOS_CURRENT_EXECUTION or something instead of using 1, so it's more clear what's going on? Then also use that around line 540: `rtos->thread_details->threadid = 1;`